open-metadata-implementation/access-services/asset-lineage/asset-lineage-server/src/main/java/org/odpi/openmetadata/accessservices/assetlineage/handlers/ContextHandler.java

```diff
@@ -1,85 +1,97 @@
 
-public class AssetContextBuilder {
+public class ContextHandler {
 
-
-    private OMRSMetadataCollection metadataCollectionForSearch;
     private Converter converter = new Converter();
     private TypeDefGallery allTypes = new TypeDefGallery();
+    private static final String guidParameter = "guid";
+    private String serviceName;
     private String serverName;
-    private ExceptionHandler exceptionUtil;
-    private OMRSAuditLog auditLog;
-
-    public AssetContextBuilder(OMRSAuditLog auditLog){
-        this.auditLog = auditLog;
-        this.exceptionUtil = new ExceptionHandler(auditLog);
-
+    private RepositoryHandler repositoryHandler;
+    private OMRSRepositoryHelper repositoryHelper;
+    private InvalidParameterHandler invalidParameterHandler;
+
+    /**
+     * Construct the discovery engine configuration handler caching the objects
+     * needed to operate within a single server instance.
+     *
+     * @param serviceName             name of the consuming service
+     * @param serverName              name of this server instance
+     * @param invalidParameterHandler handler for invalid parameters
+     * @param repositoryHelper        helper used by the converters
+     * @param repositoryHandler       handler for calling the repository services
+     */
+    public ContextHandler(String serviceName,
+                          String serverName,
+                          InvalidParameterHandler invalidParameterHandler,
+                          OMRSRepositoryHelper repositoryHelper,
+                          RepositoryHandler repositoryHandler) {
+        this.serviceName = serviceName;
+        this.serverName = serverName;
+        this.invalidParameterHandler = invalidParameterHandler;
+        this.repositoryHelper = repositoryHelper;
+        this.repositoryHandler = repositoryHandler;
     }
 
-    public AssetContext getAssetContext(String serverName, String userId, String assetId){
-        AssetContext assetContext = new org.odpi.openmetadata.accessservices.assetlineage.model.AssetContext();
+    public AssetContext getAssetContext(String serverName, String userId, String GUID) throws InvalidParameterException, FunctionNotSupportedException, PropertyServerException, PropertyErrorException, EntityProxyOnlyException, org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException, EntityNotKnownException, TypeDefNotKnownException, PagingErrorException, UserNotAuthorizedException, TypeErrorException, org.odpi.openmetadata.frameworks.connectors.ffdc.UserNotAuthorizedException, RepositoryErrorException {
+        AssetContext assetContext = new AssetContext();
 
-        try {
-            setMetadataRepositoryDetails(serverName, userId);
-            EntityDetail entityDetail = getEntityDetails(serverName, userId, assetId);
-            Map<String, List<Connection>> knownAssetConnection = new HashMap<>();
+        this.allTypes = repositoryHandler.getMetadataCollection().getAllTypes(userId);
```

Comment:
@mandy-chessell  I have added a getMetadataCollection() to the RepositoryHandler class, because I could not find an equivalent for getAllTypes(userId) within RepositoryHandler itself.